File: apps/mobile_flutter/lib/app/app_load_state.dart
enum AppLoadStatus {
  loading,
  connected,
  fallback,
}

class AppLoadState {
  const AppLoadState({
    required this.status,
    required this.message,
    this.details,
  });

  const AppLoadState.loading()
      : status = AppLoadStatus.loading,
        message = 'Connecting to local backend',
        details = null;

  const AppLoadState.connected()
      : status = AppLoadStatus.connected,
        message = 'Backend connected',
        details = null;

  const AppLoadState.fallback([this.details])
      : status = AppLoadStatus.fallback,
        message = 'Offline seed mode';

  final AppLoadStatus status;
  final String message;
  final String? details;

  bool get isLoading => status == AppLoadStatus.loading;
  bool get isFallback => status == AppLoadStatus.fallback;
}

File: apps/web_pwa/src/app/appLoadState.ts
// Mirrors apps/mobile_flutter/lib/app/app_load_state.dart.
export type AppLoadStatus = 'loading' | 'connected' | 'fallback';

export interface AppLoadState {
  status: AppLoadStatus;
  message: string;
  details?: string;
}

export const loadingState: AppLoadState = {
  status: 'loading',
  message: 'Connecting to local backend',
};

export const connectedState: AppLoadState = {
  status: 'connected',
  message: 'Backend connected',
};

export function fallbackState(details?: string): AppLoadState {
  return { status: 'fallback', message: 'Offline seed mode', details };
}

export function isLoading(state: AppLoadState): boolean {
  return state.status === 'loading';
}

export function isFallback(state: AppLoadState): boolean {
  return state.status === 'fallback';
}
